### ISSUE TITLE
Negation of rules in salt.states.iptables with "iptables --out-interface" not working

### ISSUE THREAD

**Issue by maltris** [saltstack/salt#43502]
### Description of Issue/Question
When running the function pasted in "### Setup", Salt generates a wrong rule:

```
----------
          ID: iptables-block
    Function: iptables.insert
      Result: False
     Comment: Failed to set iptables rule for iptables-block.
              Attempted rule was -p tcp -m multiport ! --dports 80,443 ! --out-interface ! br-+ --in-interface br-+ --jump DROP
     Started: 13:28:42.437708
    Duration: 64.339 ms
     Changes: 
```

Manual run according to the error message:
```
iptables -I DOCKER-ISOLATION -p tcp -m multiport ! --dports 80,443 ! --out-interface ! br-+ --in-interface br-+ --jump DROP
Bad argument `br-+'
Try `iptables -h' or 'iptables --help' for more information.
echo $?
2
```

Manual run of the correct command (as it should be):
```
iptables -I DOCKER-ISOLATION -p tcp -m multiport ! --dports 80,443 ! --out-interface br-+ --in-interface br-+ --jump DROP
echo $?
0
```

### Setup
```
iptables-block:
  iptables.insert:
    - family: ipv4
    - position: 1
    - table: filter
    - chain: DOCKER-ISOLATION
    - jump: DROP
    - match: multiport
    - dports:
        - '! 80'
        - '! 443'
    - proto: tcp
    - in-interface: 'br-+'
    - out-interface: '! br-+'
    - save: True
```

### Steps to Reproduce Issue
Run the above function/commands.

### Versions Report
```
Salt Version:
           Salt: 2017.7.1
 
Dependency Versions:
           cffi: 1.5.2
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: 0.24.0
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.24.0
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: ISO-8859-1
        machine: x86_64
        release: 4.4.0-92-generic
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by garethgreenaway** [comment]
Looks like it's doubling up the `!` for the out bound interface and also including it in the wrong spot.

**Comment by garethgreenaway** [comment]
@maltris This has been fixed on in the 2017.7 branch and will be available in the 2017.7.2 release.
